clang/test/CodeGenSYCL/code_location.cpp

```diff
@@ -0,0 +1,24 @@
+// RUN: %clang_cc1 -fsycl-is-device -internal-isystem -sycl-std=2020 -fsycl-int-header=%t.h %s -o %t.out
+// RUN: FileCheck -input-file=%t.h %s
+
+#include "Inputs/sycl.hpp"
+
+int main() {
+  cl::sycl::queue q;
+  q.submit([&](cl::sycl::handler &h) { h.single_task([] {}); });
+  q.submit([&](cl::sycl::handler &h) { h.single_task<class KernelName>([]() {}); });
+  return 0;
+}
+
+// CHECK: template <> struct KernelInfoData<'_', 'Z', 'T', 'S', 'Z', 'Z', '4', 'm', 'a', 'i', 'n', 'E', 'N', 'K', 'U', 'l', 'R', 'N', '2', 'c', 'l', '4', 's', 'y', 'c', 'l', '7', 'h', 'a', 'n', 'd', 'l', 'e', 'r', 'E', 'E', '_', 'c', 'l', 'E', 'S', '2', '_', 'E', 'U', 'l', 'v', 'E', '_'> {
+// CHECK: static constexpr const char* getFileName() { return {{.*}}/clang/test/CodeGenSYCL/code_location.cpp; }
```

Comment:
Should we be emitting the path? I am not sure if that is some sort of security issue.

Reply:
It seemed that the decision was to let the consumers know their full path/underlying tooling; hence  guarding these methods or what they return with NDEBUG? @alexbatashev @erichkeane

Reply:
I suspect we don't want to emit the whole path in this case, because it is NOT opt-in.  I think you're right about it being a security concern.